### PR TITLE
Refactor connection

### LINKED ISSUE #1039: No backpressure on pubsub and monitor leads to piling up messages
Same effect as #1034 (memory overflow), though not a leak

With pubsub, if the producer is much faster than the consumer, the messages pile up in `Connection::dispatch_q_` as `SendMsgVecAsync` is fully async and does not provide any backpressure

### PR DESCRIPTION
**Pull request by dranikpg**
This PR fixes #1039 and:
* refactors connection - first commits have no functional changes
* limits max size of dispatch queue size - last commit

Please look at them on a per-commit basis, because I also did some refactoring.

### CODE REVIEW

**Comment by dranikpg** [comment]
Need so see what redis actually does. A memory limit saves you from an OOM, but with small messages the lag can still become gigantic

File: src/facade/dragonfly_connection.h [hunk 1]
```diff
@@ -162,7 +168,11 @@ class Connection : public util::Connection {
 
  private:
   std::deque<RequestPtr> dispatch_q_;  // dispatch queue
-  dfly::EventCount evc_;               // dispatch queue waker
+  dfly::EventCount dispatch_q_ec;      // dispatch q waker
```
Comment (romange) [inline src/facade/dragonfly_connection.h:171]: private data members should have underscores at the end

**Review by romange — COMMENTED** [review]
LGTM overall

**Comment by dranikpg** [comment]
I've removed the new stuff and left this as a plain refactoring PR, because I just realized it'll add too much complexity

**Review by romange — APPROVED** [review]

**Pull request merged**